### PR TITLE
Katana on the sabre.io website

### PR DESCRIPTION
**Pull request by evert**
Note that I stripped all custom design from the initial proposal. I can easily get it back, but it was a lot easier to start with content and go from there, rather than the other way round.

### CODE REVIEW

**Comment by evert** [comment]
In the same vain as the other katana PR, I don't consider this completed and final, but unlike that branch I feel that this is better than 'nothing at all', so unless anyone else disagrees, I'd like merge this and deploy it asap.

**Comment by Hywan** [comment]
We should update the “login” screenshot.

File: source/katana/goals.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+title: Development goals 
+product: katana
+layout: default
+---
+
+While sabre/dav has obtained a decent marketshare for cal-, card-, and webdav
+servers, traditionally it's always been intended to be a development library.
```
Comment (Hywan) [inline source/katana/goals.md]: `Cal-, Card- and WebDAV servers`

File: source/katana/goals.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+It's our goal to make the admin interface an independent component, so it can
+also be used for any other sabre/dav server.
+
+
+Installer design goals
+----------------------
+
+The installer needs to help with getting people started getting their server
+up and running. Traditionally it was only ever possible to set up sabre/dav
+by copying files, and you were even forced to manually run sql queries.
```
Comment (Hywan) [inline source/katana/goals.md]: `SQL`

File: source/katana/goals.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+The installer will now walk you throug the steps, in a way so that it's
```
Comment (Hywan) [inline source/katana/goals.md]: `through`

File: source/katana/goals.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+We will also add an admin interface. This admin interface should be seperable
+from the rest of the package.
+
+* The admin interface should be 100% client side. PHP must not be used.
+* It should be possible to deploy an admin interface on a static host, such
+  as S3.
+* The admin interface should be able to communicate with *any* sabre/dav
+  server.
+* The admin interface allows the admin to:
+  * manage users.
```
Comment (Hywan) [inline source/katana/goals.md]: `Manage`

File: source/katana/goals.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  * manage calendars.
```
Comment (Hywan) [inline source/katana/goals.md]: `Manage`

File: source/katana/goals.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  * manage address books.
```
Comment (Hywan) [inline source/katana/goals.md]: `Manage`

File: source/katana/goals.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  * manage files.
```
Comment (Hywan) [inline source/katana/goals.md]: `Manage`

File: source/katana/goals.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+Post 1.0 features:
+
+* It should also be able to check basic sabre/dav version information and
+  report vulnerabilities.
+* Work in a "non-admin" mode, so users can manage their own calendars even if
+  they don't have admin-related features.
+* It should be tested with some non-sabre/dav-based servers. 
+
+
```
Comment (Hywan) [inline source/katana/goals.md]: Remove trailing empty lines.

File: source/katana/index.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
+---
+layout: default
+sidebar: none
+product: katana
+---
+
+sabre/katana is the codename for our next-generation cal, card- and webdav
```
Comment (Hywan) [inline source/katana/index.md]: `Cal-, Card- and WebDAV`

File: source/katana/index.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+<img src="{{ site.url }}/img/screen-katana-admin.png" alt="Log in screen for sabre/katana">
+
+
+Features
+--------
+
+sabre/katana will support the following features:
+
+1. Web and CLI installer.
+2. CalDAV
```
Comment (Hywan) [inline source/katana/index.md]: `CalDAV.`

File: source/katana/index.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+3. CardDAV
```
Comment (Hywan) [inline source/katana/index.md]: `CardDAV.`

File: source/katana/index.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+4. WebDAV
```
Comment (Hywan) [inline source/katana/index.md]: `WebDAV.`

File: source/katana/install.md [hunk 13]
```diff
@@ -0,0 +1,63 @@
+---
+title: sabre/katana
+product: katana
+layout: default
+---
+
+This document explains how to install sabre/katana. We currently officially
+support two main installation methods:
+
+* [from a zip archive](#install-from-zip)
```
Comment (Hywan) [inline source/katana/install.md]: `ZIP`

File: source/katana/install.md [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+* [from source](#install-from-source)
+
+Unless you are a developer, you will most likely want to follow the
+installation instructions for the zip archive.
```
Comment (Hywan) [inline source/katana/install.md]: `ZIP`

File: source/katana/install.md [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+Requirements
+------------
+
+* PHP 5.5 and up
```
Comment (Hywan) [inline source/katana/install.md]: `up.`

File: source/katana/install.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+* MySQL or Sqlite database. MySQL is _highly_ recommended.
```
Comment (Hywan) [inline source/katana/install.md]: `SQLite`

File: source/katana/install.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+* Webserver running on Linux or Mac.
+
+
+Install from zip
+----------------
+
+1. Download the latest zip archive from [github][1].
```
Comment (Hywan) [inline source/katana/install.md]: `ZIP`
`Github`

File: source/katana/install.md [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+2. Unzip the archive, and place all the contents into a location that's
+   accessible through a webserver.
+3. Navigate to the installer. If you downloaded the zip archive and placed it
```
Comment (Hywan) [inline source/katana/install.md]: `ZIP` (everywhere)

File: source/katana/install.md [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+   `http://yourserver/katana/public/install.php`, but bear in mind that this
+   might be different for you.
+4. Follow the instructions in the browser.
+
+
+From source
+-----------
+
+If you are a developer, and you want to install sabre/katana via Git, make
+sure you've cloned the repository located at http://github.com/fruux/sabre-katana/
```
Comment (Hywan) [inline source/katana/install.md]: No link here?

File: source/katana/install.md [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+
+After that, you'll need:
+
+* [composer][2]
```
Comment (Hywan) [inline source/katana/install.md]: `Composer`

No link to NPM and Bower?

File: source/katana/install.md [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+* NPM
```
Comment (Hywan) [inline source/katana/install.md]: `NPM.`

File: source/katana/install.md [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+* Bower
```
Comment (Hywan) [inline source/katana/install.md]: `Bower.`

File: source/katana/install.md [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+
+to retrieve the various packages that are used in sabre/katana.
+
+If you have all these package managers, run:
+
+    composer install
+    bowser install
```
Comment (Hywan) [inline source/katana/install.md]: Wrong.
```sh
$ make install
```

File: source/katana/upgrade.md [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+title: sabre/katana
+product: katana
+layout: default
+---
+
+Before updating, it's highly recommended to make a backup of your installation
+directory and database.
+
+To update your sabre/katana to the latest version, grab the latest version
+from [GitHub][1].
```
Comment (Hywan) [inline source/katana/upgrade.md]: `Github` (like proposed above)

File: source/katana/upgrade.md [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+
+After that, you can simply unzip the archive and overwrite all existing files
+and directories *except* the `data/` directory. If you overwrite your `data/`
+directory, you will effectively delete all existing data.
```
Comment (Hywan) [inline source/katana/upgrade.md]: Why not giving the command lines here? What about the PHAR :smile:?

File: source/license.md [hunk 26]
```diff
@@ -6,7 +6,7 @@ nocomments: true
 License
 =======
 
-All the sabre-projects use the three-clause BSD-license.
+All the sabre-projects, with the exception of sabre/katana, use the three-clause BSD-license.
```
Comment (Hywan) [inline source/license.md]: :+1:

**Comment by Hywan** [comment]
</review>

File: source/katana/install.md (hunk 17 above)
Comment (DominikTo) [inline source/katana/install.md]: The official brand is actually `GitHub`. :-)

**Comment by evert** [comment]
I agree with most of the suggested changes. Didn't have a link to NPM because it was written on a train with no internet ;)

The reason I reduced both the installation and upgrading documentation is simplicity. The goal really is to cater to the lowest common denominator. I don't want to assume a command line, I just assume a guy on a windows machine with an ftp client ;)

I initially made those the instructions in the top of the page, and kept the rest... but then I realized that it wasn't really needed to keep the other instructions. I don't want people to have to make the decision which instructions they want to follow. Whether updating uses a phar or a zip is completely irrelevant to the user, with the exception that the phar installation might not work.

**Comment by evert** [comment]
The other part is also that I think we should not officially support any other installation method other than:

1. this manual way
2. install from source
3. operating system-specific packages

If we have a matrix of potential installation methods, we also have a matrix of potential bugs and this greatly increases the number of problems we will run into, which will also affect the perception of stability people will have. I want to pick 1 single way of doing things and just document and support that. Anything else is 'you're doing it wrong'.

**Comment by DominikTo** [comment]
Lets at least add the Katana logo! :)

**Comment by evert** [comment]
k!

**Comment by evert** [comment]
Cool to deploy?

**Comment by DominikTo** [comment]
It's a start we can iterate on. :+1:

**Comment by Hywan** [comment]
Agree for installation and update instructions :-).

**Pull request merged**